src/main/java/duke/util/Storage.java

```diff
@@ -0,0 +1,110 @@
+package duke.util;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+/**
+ * Represents the storage function of the program. Deals with loading and saving tasks to a file.
+ */
+public class Storage {
+    private final String filePath;
+    private final String fileName;
+
+    /**
+     * Class constructor.
+     * Creates a new file if it does not already exist.
+     *
+     * @param filePath The filepath to the text file.
+     * @param fileName The name of the text file.
+     */
+    public Storage(String filePath, String fileName) {
+        this.filePath = filePath;
+        this.fileName = fileName;
+        createFileAndFolderIfDoesNotExist();
+    }
+
+    /**
+     * Creates the text file and folder if they do not already exist.
+     */
+    public void createFileAndFolderIfDoesNotExist() {
+        File folder = new File(filePath);
+        try {
+            if (!folder.exists()) {
+                folder.mkdir();
+            } else {
+                File file = new File(filePath + fileName);
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println("Unable to create file to save tasks.");
+        }
```

Comment:
I think this part can be modified to make `filePath` more accurate; as of now `filePath` is being treated more like the path of the directory that the file is located in. Perhaps could make use of `getParentFile()` and `mkdirs()`? i.e. from

```java
File folder = new File(filePath);
try {
    if (!folder.exists()) {
        folder.mkdir();
    } else {
        File file = new File(filePath + fileName);
        file.createNewFile();
    }
} catch (IOException e) {
    System.out.println("Unable to create file to save tasks.");
}
```
to
```java
File file = new File(filePath);
try {
    if (!file.exists()) {
        file.getParentFile().mkdirs();
        file.createNewFile();
    }
} catch (IOException e) {
    System.out.println("Unable to create file to save tasks.");
}
```